src/libaktualizr-posix/ipuptanesecondary.cc

```diff
@@ -1,32 +1,75 @@
-#include "ipuptanesecondary.h"
+#include <arpa/inet.h>
+#include <netinet/tcp.h>
+
 #include "asn1/asn1_message.h"
 #include "der_encoder.h"
+#include "ipuptanesecondary.h"
 #include "logging/logging.h"
 
 #include <memory>
 
 namespace Uptane {
 
-PublicKey IpUptaneSecondary::getPublicKey() {
-  LOG_INFO << "Getting the public key of a secondary";
-  Asn1Message::Ptr req(Asn1Message::Empty());
+std::shared_ptr<Uptane::SecondaryInterface> IpUptaneSecondary::create(const std::string& address, unsigned short port) {
+  LOG_INFO << "Creating IP Secondary: " << address << ":" << port << "...";
 
-  req->present(AKIpUptaneMes_PR_publicKeyReq);
+  EcuSerial serial{"serial"};
+  HardwareIdentifier hw_id{"hw_id"};
```

Comment:
I'd suggest initializing these to empty strings. Otherwise it might be confusing in error conditions. Especially for testing, we often use dummy values that might be easily confused with these defaults.

Reply:
I tried, but there is an input string length verification in EcuSerial ctor what doesn't allow me to specify an empty string.
Can I remove it as someone removed it for HardwareIdentifier

Reply:
I think I remember running into that before, and somehow I was able to work around it. That seems less possible here. We'd need to do a quick check that removing that restriction won't open up any holes, but it's probably fine.

Reply:
Looks like remove it is fine, at least tests passes on my local env.

Reply:
I was fairly sure the tests would pass, but I mean we should look at where EcuSerials are instantiated and check if we need to add any new checks for non-emptiness in those places.